ingestion/src/metadata/ingestion/lineage/models.py

```diff
@@ -0,0 +1,91 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+Models related to lineage parsing
+"""
+from enum import Enum
+from typing import Dict
+
+from metadata.generated.schema.entity.services.connections.database.bigQueryConnection import (
+    BigqueryType,
+)
+from metadata.generated.schema.entity.services.connections.database.clickhouseConnection import (
+    ClickhouseType,
+)
+from metadata.generated.schema.entity.services.connections.database.db2Connection import (
+    Db2Type,
+)
+from metadata.generated.schema.entity.services.connections.database.deltaLakeConnection import (
+    DeltaLakeType,
+)
+from metadata.generated.schema.entity.services.connections.database.hiveConnection import (
+    HiveType,
+)
+from metadata.generated.schema.entity.services.connections.database.mysqlConnection import (
+    MySQLType,
+)
+from metadata.generated.schema.entity.services.connections.database.oracleConnection import (
+    OracleType,
+)
+from metadata.generated.schema.entity.services.connections.database.postgresConnection import (
+    PostgresType,
+)
+from metadata.generated.schema.entity.services.connections.database.redshiftConnection import (
+    RedshiftType,
+)
+from metadata.generated.schema.entity.services.connections.database.snowflakeConnection import (
+    SnowflakeType,
+)
+from metadata.generated.schema.entity.services.connections.database.sqliteConnection import (
+    SQLiteType,
+)
+
+
+class Dialect(Enum):
+    """
+    Supported dialects by sqllineage
+    """
+
+    ANSI = "ansi"
+    ATHENA = "athena"
+    BIGQUERY = "bigquery"
+    CLICKHOUSE = "clickhouse"
+    DB2 = "db2"
+    DUCKDB = "duckdb"
+    EXASOL = "exasol"
+    HIVE = "hive"
+    MATERIALIZE = "materialize"
+    MYSQL = "mysql"
+    ORACLE = "oracle"
+    POSTGRES = "postgres"
+    REDSHIFT = "redshift"
+    SNOWFLAKE = "snowflake"
+    SOQL = "soql"
+    SPARKSQL = "sparksql"
+    SQLITE = "sqlite"
+    TERADATA = "teradata"
+    TSQL = "tsql"
+
+
+MAP_CONNECTION_TYPE_DIALECT: Dict[str, Dialect] = {
+    str(MySQLType.Mysql.value): Dialect.ATHENA,
```

Comment:
could we add a comment on why this mapping?

Reply:
Moved the logic into `ConnectionTypeDialectMapper` class.